python/lsst/source/injection/inject_base.py

```diff
@@ -0,0 +1,636 @@
+        default=100,
+        optional=True,
+    )
+    subset_by_select = Field[bool](
+        doc="If True, limit the input catalog to a subset of sources flagged as True in the *col_select* "
+        "column.",
+        default=False,
+    )
+
+    # General configuration options.
+    mask_plane_name = Field[str](
+        doc="Name assigned to the injected mask plane which is attached to the output exposure.",
+        default="INJECTED",
+    )
+    calib_flux_radius = Field[float](
+        doc="Aperture radius (in pixels) that was used to define the calibration for this image+catalog. "
+        "This will be used to produce the correct instrumental fluxes within the radius. "
+        "This value should match that of the field defined in ``slot_CalibFlux_instFlux``.",
+        default=12.0,
+    )
+    fits_alignment = ChoiceField[str](
+        doc="How should injections from FITS files be aligned?",
+        dtype=str,
+        allowed={
+            "wcs": (
+                "Input image will be transformed such that the local WCS in the FITS header matches the "
+                "local WCS in the target image. I.e., North, East, and angular distances in the input image "
+                "will match North, East, and angular distances in the target image."
+            ),
+            "pixel": (
+                "Input image will **not** be transformed. Up, right, and pixel distances in the input image "
+                "will match up, right and pixel distances in the target image."
+            ),
+        },
+        default="pixel",
+    )
+    stamp_prefix = Field[str](
+        doc="String to prefix to the entries in the *col_stamp* column, for example, a directory path.",
+        default="",
+    )
+    draw_size_scale = Field[float](
+        doc="Multiplicative scale factor applied to the size of the injected box drawn for each source.",
+        default=1.0,
+    )
+    draw_size_max = Field[int](
+        doc="Maximum allowed pixel size of the injected box drawn for each source. "
+        "Clipped to this value if set greater than zero.",
+        default=10000,
+    )
+
+    # Custom column names.
+    col_ra = Field[str](
+        doc="Column name for right ascension (in degrees).",
+        default="ra",
+    )
+    col_dec = Field[str](
+        doc="Column name for declination (in degrees).",
+        default="dec",
+    )
+    col_mag = Field[str](
+        doc="Column name for magnitude.",
+        default="mag",
+    )
+    col_source_type = Field[str](
+        doc="Column name for the source type used in the input catalog. Must match one of the surface "
+        "brightness profiles defined by GalSim.",
+        default="source_type",
+    )
+    col_select = Field[str](
+        doc="Column name for boolean flags used to identify which sources to inject. "
+        "Will only be applied when *subset_by_select* is True.",
+        default="select",
+    )
+    col_stamp = Field[str](
```

Comment:
Two points come to mind:

1. During testing, whilst collaborating with another developer, I found I was not able to user their injection catalog that they sent to me as-is without needing to modify the `stamp` column. They had hard-coded the full path to these FITS files on their local disk, and I was not able to run with that catalog out of the box.
2. For large catalogs, an absolute file path encodes a lot of duplicated information. It becomes easier to parse the catalog if the `stamp` column only encodes the local stub, leaving the local root to be specified elsewhere.

With that said, there's also currently nothing stopping an end user from continuing to put the full path into the `stamp` column right now and leaving `stamp_prefix` as its default value (`""`). This will append nothing to the front of each filename, preserving the behavior you suggest.